Update shopping-cart-product-line.tpl

jdperea opened the pull request:
In mobile the users can't see if is aviable or out stock

jdperea committed:
Update shopping-cart-product-line.tpl

In mobile the users can't see if is aviable or out stock

gskema commented:
Can comment what exactly did you change? It's hard to see what has been changed in the diff.

jdperea commented:
Line 10-> I added all the included between the span tags original content from line 13 setting the class visible-xs to the span class.

`<span class="visible-xs label{if $product.quantity_available <= 0 && isset($product.allow_oosp) && !$product.allow_oosp} label-danger{elseif $product.quantity_available <= 0} label-warning{else} label-success{/if}">{if $product.quantity_available <= 0}{if isset($product.allow_oosp) && $product.allow_oosp}{if isset($product.available_later) && $product.available_later}{$product.available_later}{else}{l s='In Stock'}{/if}{else}{l s='Out of stock'}{/if}{else}{if isset($product.available_now) && $product.available_now}{$product.available_now}{else}{l s='In Stock'}{/if}{/if}</span>{if !$product.is_virtual}{hook h="displayProductDeliveryTime" product=$product}{/if}`

Line 13-> I changed from <td class="cart_avail"> to <td class="cart_avail hidden-xs">
From this `<td class="cart_avail">`
To this `<td class="cart_avail hidden-xs">`

jdperea commented on themes/community-theme-16/shopping-cart-product-line.tpl:
```diff
@@ -7,10 +7,10 @@
   </td>
   {if $PS_STOCK_MANAGEMENT}
-    <td class="cart_avail"><span class="label{if $product.quantity_available <= 0 && isset($product.allow_oosp) && !$product.allow_oosp} label-danger{elseif $product.quantity_available <= 0} label-warning{else} label-success{/if}">{if $product.quantity_available <= 0}{if isset($product.allow_oosp) && $product.allow_oosp}{if isset($product.available_later) && $product.available_later}{$product.ava ...
+    <td class="cart_avail hidden-xs"><span class="label{if $product.quantity_available <= 0 && isset($product.allow_oosp) && !$product.allow_oosp} label-danger{elseif $product.quantity_available <= 0} label-warning{else} label-success{/if}">{if $product.quantity_available <= 0}{if isset($product.allow_oosp) && $product.allow_oosp}{if isset($product.available_later) && $product.available_later}{$p ...
```
Added to `<td>` the class hidden-xs

jdperea commented on themes/community-theme-16/shopping-cart-product-line.tpl:
```diff
@@ -7,10 +7,10 @@
     <p class="product-name"><a href="{$link->getProductLink($product.id_product, $product.link_rewrite, $product.category, null, null, $product.id_shop, $product.id_product_attribute, false, false, true)|escape:'html':'UTF-8'}">{$product.name|escape:'html':'UTF-8'}</a></p>
     {if $product.reference}<small class="cart_ref">{l s='SKU'}{$smarty.capture.default}{$product.reference|escape:'html':'UTF-8'}</small>{/if}
-    {if isset($product.attributes) && $product.attributes}<small><a href="{$link->getProductLink($product.id_product, $product.link_rewrite, $product.category, null, null, $product.id_shop, $product.id_product_attribute, false, false, true)|escape:'html':'UTF-8'}">{$product.attributes|@replace: $smarty.capture.sep:$smarty.capture.default|escape:'html':'UTF-8'}</a></small>{/if}
+    {if isset($product.attributes) && $product.attributes}<small><a href="{$link->getProductLink($product.id_product, $product.link_rewrite, $product.category, null, null, $product.id_shop, $product.id_product_attribute, false, false, true)|escape:'html':'UTF-8'}">{$product.attributes|@replace: $smarty.capture.sep:$smarty.capture.default|escape:'html':'UTF-8'}</a></small>{/if}<span class="visible ...
```
I added all the included between the span tags original content from line 13 setting the class visible-xs to the span class.

gskema commented:
Hello,

sorry for not responding for so long. I formatted your PR and checked the difference:

http://www.mergely.com/iNiM17b1/

Some things I noticed:
1. Label inside `.cart_description` no longer respects `PS_STOCK_MANAGEMENT` option.
2. By using `hidden-xs` we are hiding the `td`s, but the `th` is still present - table gets mangled.
3. Labels are actually visible in XS, because we are using `.table-responsive` wrapper - you need to scroll sideways. The rest of the columns are to he right as well.

Responsive tables are hard and messy so we resorted to this simple solution, so that the developer can redo it in whatever way he wants.

Since the labels are working find the way they are, I'm closing this.

jdperea commented:
Ok, thank you
